### PR TITLE
chore: update landingpage

### PR DESCRIPTION
**Pull request by georgiee**
This will update the landingpage:
+ Fix service keys (some rebase error on the keys)
+ Split from the gatsby page + create stories for each section
+ Provide some image spacer solution (which we need to switch to gatsby images quickly for some image optimizations)
+ Provide the blog section

### CODE REVIEW

File: src/@types/blog.d.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export interface BlogPostTeaser {
```
Comment (georgiee) [inline src/@types/blog.d.ts]: this is shared across two locations so I put them in here, I'm not sure how to handle shared types in general right now.

File: src/new-components/i18n-helpers.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { useTranslation } from 'gatsby-plugin-react-i18next';
+import { format, parseISO } from 'date-fns';
+import { enGB, de } from 'date-fns/locale';
+/**
+ * A custom  hook to read the current language
+ * and return a formatter (date-fns) with the correct locale set
+ * with the default being `enGB` for any unknown language.
+ */
+export const useLocalFormat = (dateFormat) => {
```
Comment (georgiee) [inline src/new-components/i18n-helpers.ts]: First I had a format factory function with a lang parameter. I recognized that the react way is to use a custom hook which acts as the same factory anyway. feels good 👍

File: src/page-building/landingpage/blog.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+              cover={image && <GatsbyImage alt="" image={image} />}
+            >
+              {item.frontmatter.teaserText}
+            </Teaser>
+          );
+        })}
+      </TeaserGrid>
+    </>
+  );
+};
```
Comment (georgiee) [inline src/page-building/landingpage/blog.tsx]: shiny new blog section for the landingpage. I don't like how the teaser render the actual blog content.
the design does not work for us. 

+ We don't need the kicker
+ The kicker conflicts with the timestamp easily (same line)
+ I don't have a place to easily show the author which we might want to do.

Screen Shot 2021-12-10 at 14 13 41

File: src/page-building/landingpage/career.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+      <TeaserGrid>
+        {positions.map((item) => (
+          <Teaser key={item.id} title={item.name} linkTo={item.fields?.path}>
+            {item.short}
+          </Teaser>
+        ))}
+      </TeaserGrid>
+    </>
+  );
+};
```
Comment (georgiee) [inline src/page-building/landingpage/career.tsx:34]: This section existed before inline and I created a separate file now side by side to the stories.

File: src/page-building/landingpage/landingpage.stories.mdx [hunk 5]
```diff
@@ -0,0 +1,48 @@
+import { ArgsTable, Canvas, Meta, Story } from '@storybook/addon-docs'; 
+import { Career } from './career';
+import { Service } from './service'; import { Blog } from './blog'; import { GatsbyImage, getImage } from 'gatsby-plugin-image';
+
+<Meta title="Pages/Landingpage" component={{ Image }} />
+
+# Landingpage
+The landingpage page consists of various sections you can see here
```
Comment (georgiee) [inline src/page-building/landingpage/landingpage.stories.mdx:8]: The minimum story set of the landingpages. Yoy can now see the various sections from the usage perspective which is quite nice.

File: src/page-building/landingpage/landingpage.tsx [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+  }}
+
+`;
+const ImageSeparator = ({ image }) => {
+  return (
+    <FullWidthContainer>
+      <ImageFull src={image} />
+    </FullWidthContainer>
+  );
+};
```
Comment (georgiee) [inline src/page-building/landingpage/landingpage.tsx:73]: This is my try to create the required image separators. I will refactor this so it's just inline in the landingpage component.

File: src/page-building/landingpage/landingpage.tsx [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+
+export interface LocalesQuery {
+  edges: {
+    node: {
+      ns: string;
+      language: string;
+      data: string;
+    };
+  }[];
+}
```
Comment (georgiee) [inline src/page-building/landingpage/landingpage.tsx]: I think that's obsolete?

File: src/page-building/landingpage/landingpage.tsx [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+      light={true}
+      hero={
+        <>
+          <Aurora />
+          <HeroContainer>
+            <IndexPageTitle>Satellytes</IndexPageTitle>
+            <IndexPageSubTitle as="h2">
+              {t('main.description')}
+            </IndexPageSubTitle>
+          </HeroContainer>
```
Comment (georgiee) [inline src/page-building/landingpage/landingpage.tsx:94]: We need a "landingpage hero" component not to hack this together all the time. I will work on that when working on the heros in general (we need a gatsby image based hero with cover functionality)

File: src/page-building/landingpage/landingpage.tsx [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+  return (
+    <FullWidthContainer>
+      <ImageFull src={image} />
+    </FullWidthContainer>
+  );
+};
+
+const FullWidthContainer = styled(ContentBlockContainer)`
+  grid-column: -1/1;
+`;
```
Comment (georgiee) [inline src/page-building/landingpage/landingpage.tsx:77]: I did not find a container to spread across the grid. I though we had one before? So I keep it local for now.

File: src/pages/index.tsx [hunk 10]
```diff
@@ -71,112 +15,21 @@ interface IndexPageProps {
...
-              }
-            >
-              {t('main.services.teasers.third.text')}
-            </Teaser>
-          </TeaserGrid>
-        </ContentBlockContainer>
-
-        <CareerSection positions={jobPositions} />
-      </LayoutV2>
+      <Landingpage positions={jobPositions} posts={blogPosts} />
```
Comment (georgiee) [inline src/pages/index.tsx:28]: Yes, I can breathe again also here.

File: src/pages/index.tsx [hunk 11]
```diff
@@ -209,5 +62,32 @@ export const IndexPageQuery = graphql`
         }
       }
     }
+
+    allMarkdownRemark(
```
Comment (georgiee) [inline src/pages/index.tsx:62]: we query for the blog posts but then pass the into the actual page component. This keeps things cleaned up 👍

File: src/@types/blog.d.ts (hunk 1 above)
Comment (feedm3) [reply src/@types/blog.d.ts]: `@types` folder is for global definition files, isn't it? so not for "regular shared interfaces", but I would need to dig deeper into that topic to make a technically correct statement. For now it's ok to have it here.

In other projects, we are using `src/types.ts`.

File: src/new-components/i18n-helpers.ts (hunk 2 above)
Comment (feedm3) [reply src/new-components/i18n-helpers.ts]: Typo alarm 🚨 

`useLocalFormat` -> `useLocaleFormat`
`getLocal` -> `getLocale`

**Comment by georgiee** [comment]
amended and force pushed to trigger a fresh check as the PR checks hung.

File: src/page-building/landingpage/landingpage.stories.mdx (hunk 5 above)
Comment (feedm3) [reply src/page-building/landingpage/landingpage.stories.mdx:8]: Nice 👍 

Depending on the data, I can think of quite complex mock data we need to take care of.

File: src/page-building/landingpage/landingpage.tsx (hunk 9 above)
Comment (feedm3) [reply src/page-building/landingpage/landingpage.tsx:77]: `FullWidth` in `layout-v2.tsx`.

Implementation is a bit different:
```
export const FullWidth = styled.div`
  width: 100%;
  grid-column: 1 / 4;
`;
```

I have hidden this very well 🕵️

**Review by feedm3 — APPROVED** [review]
Nothing critical found, good work 👍

File: src/@types/blog.d.ts (hunk 1 above)
Comment (georgiee) [reply src/@types/blog.d.ts]: oh you are right. I had some weird feeling doing so. I will create `types.ts`

File: src/new-components/i18n-helpers.ts (hunk 2 above)
Comment (georgiee) [reply src/new-components/i18n-helpers.ts]: thanks for the catch! fixed

File: src/page-building/landingpage/landingpage.stories.mdx (hunk 5 above)
Comment (georgiee) [reply src/page-building/landingpage/landingpage.stories.mdx:8]: Yes, but we can place them in `mock.ts` files or just don't showcase them if not suited.

File: src/page-building/landingpage/landingpage.tsx (hunk 9 above)
Comment (georgiee) [reply src/page-building/landingpage/landingpage.tsx:77]: that implementation needs to know the amount of columns. negative columns do not need that. let's keep both an I'm happy to refactor the layout + containers with you once we have the important pieces together 🙌

**Pull request merged**